sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1260,26 +1260,51 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
     CaseWhen(branches, Option(ctx.elseExpression).map(expression))
   }
 
+  private def canApplyRegex(ctx: ParserRuleContext): Boolean = withOrigin(ctx) {
```

Comment:
Please add a comment above this function to explain why `regex` can be applied under `NamedExpression` only. Thanks!